src/values.js

```diff
@@ -115,7 +115,14 @@ Ref.prototype.inspect = function() {
  * @returns {boolean}
  */
 Ref.prototype.equals = function(other) {
-  return other instanceof Ref && this.value === other.value;
+  return (other instanceof Ref) &&
+    this.id === other.id &&
+    (this.class instanceof Ref) &&
+    (other.class instanceof Ref) &&
+    this.class.equals(other.class) &&
+    (this.database instanceof Ref) &&
```

Comment:
I think this will still fail. I was told `class` and `database` are not required fields.

```
$ curl -u secret: http://localhost:8443/ -H 'X-FaunaDB-API-Version: 2.1' -d '{ "database": "db" }'
{
  "resource": {
    "@ref": { "id": "db", "class": { "@ref": { "id": "databases" } } }
  }
}
```

Reply:
gah. sorry. I'm jumping around too much.